tractor/trionics/_supervisor.py

```diff
@@ -0,0 +1,256 @@
+                    # started task passes back via `.started()` ?
+                    # seems like that won't work with also returning
+                    # a "task handle"?
+                    task_status.started()
+
+                    # invoke underlying func now that cs is entered.
+                    outcome = await acapture(async_fn, *args)
+
+                    # TODO: instead, mngr.send(outcome) so that we don't
+                    # tie this `.start_soon()` impl to the
+                    # `MaybeOutcome` type? Use `Outcome.send(mngr)`
+                    # right?
+                    maybe_outcome._set_outcome(outcome)
+
+        await n.start(_start_wrapped_in_scope)
+        assert to_return is not None
+
+        # TODO: better way to concat the values delivered by the user
+        # provided `.scope_manager` and the outcome?
+        return tuple([maybe_outcome] + to_return)
+
+
+# TODO: maybe just make this a generator with a single yield that also
+# delivers a value (of some std type) from the yield expression?
+# @trio.cancel_scope_manager
+@cm
+def add_task_handle_and_crash_handling(
+    nursery: Nursery,
+    scope: CancelScope,
+    maybe_outcome: MaybeOutcome,
+
+) -> Generator[None, list[Any]]:
+
+    cs: CancelScope = CancelScope()
+
+    # if you need it you can ask trio for the task obj
+    task: Task = trio.lowlevel.current_task()
+    print(f'Spawning task: {task.name}')
+
+    try:
+        # yields back when task is terminated, cancelled, returns?
+        with cs:
+            # the yielded values here are what are returned to the
+            # nursery's `.start_soon()` caller
+
+            # TODO: actually make this work so that `MaybeOutcome` isn't
+            # tied to the impl of `.start_soon()` on our custom nursery!
+            task_outcome: Outcome = yield [cs]
+
+    except Exception as err:
+        # Adds "crash handling" from `pdbp` by entering
+        # a REPL on std errors.
+        pdbp.xpm()
+        raise
+
+
+@acm
+async def open_nursery(
+    scope_manager = None,
+    **kwargs,
+):
+    async with trio.open_nursery(**kwargs) as nurse:
+        yield ScopePerTaskNursery(
+            nurse,
+            scope_manager=scope_manager,
+        )
+
+
+async def sleep_then_err():
+    await trio.sleep(1)
+    assert 0
+
+
+async def sleep_then_return_val(val: str):
+    await trio.sleep(0.2)
+    return val
+
+
+if __name__ == '__main__':
+
+    async def main():
+        async with open_nursery(
+            scope_manager=add_task_handle_and_crash_handling,
+        ) as sn:
+            for _ in range(3):
+                outcome, cs = await sn.start_soon(trio.sleep_forever)
+
+            # extra task we want to engage in debugger post mortem.
+            err_outcome, *_ = await sn.start_soon(sleep_then_err)
+
+            val: str = 'yoyoyo'
+            val_outcome, cs = await sn.start_soon(sleep_then_return_val, val)
+            res = await val_outcome.unwrap()
```

Comment:
As already mentioned above in the comment on the method, probably a better name is warranted for this since you're actually first waiting for the outcome **then** unwrapping it's underlying value/result.